LiskHQ/lisk-sdk
# Fix transaction related dependency - Closes #3614

### What was the problem?
Transaction related logic had dependency everywhere and logic was scattered.

### How did I fix it?
- Transaction pool depends on blocks
- blocks still temporally depends on transaction pool
- removed proxy `transaction.js` and move API related logic to node status (only used for node status endpoint)
- Removed dependency of blocks (_getCurrentContext) from process_transactions
- Changed process transactions to function
- Directly expose transaction pool as module
- Skipped/replaced dapp related/using tests
- Removed all callbacks in transactions and transaction pool

### How to test it?
- Run the application
- Check the changed tests carefully
- Run the tests

### Review checklist

* The PR resolves #3614  
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated
